test/constants.ts

```diff
@@ -0,0 +1,184 @@
+import Web3, { ContractInstance } from 'web3';
+import FakeProvider from 'web3-fake-provider';
+import BigNumber from 'bignumber.js';
+
+import {
+  DeferredTransactionWrapper,
+  ITxParams,
+  TypeChainContract
+} from '../types/typechain-runtime';
+
+import { AbiDefinition } from '@0xproject/types';
+
+/**
+ * Provides a harness to test autogenerated contract types
+ */
+export class TestContract<T extends TypeChainContract> {
+  private mockWeb3: Web3;
+  private mockCustomWeb3: Web3;
+
+  private mockWeb3Contract: ContractInstance;
+  private mockCustomWeb3Contract: ContractInstance;
+
+  constructor(contractName: string, private address: string, contractCode?: string) {
+    this.mockWeb3 = new Web3(new FakeProvider());
+
+    this.mockWeb3Contract = {
+      address: address,
+      abi: require(`../node_modules/market-solidity/build/contracts/${contractName}.json`)
+    };
+
+    spyOn(this.mockWeb3.eth, 'getCode').and.callFake(
+      (addr: string, cb: (err: Error, code: string) => void) => {
+        expect(addr).toEqual(address);
+        cb(null, contractCode || '0x234');
+      }
+    );
+
+    this.setupMockContract(this.mockWeb3, this.mockWeb3Contract);
+  }
+
+  /**
+   * Creates the contract type for the specified class
+   *
+   * @param contractClass The contract type class
+   */
+  public async createContract(
+    createMethod: (web3: any, address: string | BigNumber) => Promise<T>
+  ) {
+    return createMethod(this.mockWeb3, this.address);
+  }
+
+  /**
+   * Mocks out a web3 method
+   *
+   * @param methodName The method to mock out
+   * @param expectedOrError The expected error or return value
+   * @param expectedArgs The expected arguments
+   */
+  public setupMethodSpy(methodName: keyof T, expectedOrError: Error | any, ...expectedArgs: any[]) {
```

Comment:
@ConnorChristie -- Can we hardcode `methodName` of type `string` instead of `keyof T`?